network/src/peers/peers.rs

```diff
@@ -318,7 +327,11 @@ impl Node {
 
         let basic_filter =
             |peer: &Peer| peer.address != remote_address && !self.config.bootnodes().contains(&peer.address);
-        let strict_filter = |peer: &Peer| basic_filter(peer) && peer.is_routable.unwrap_or(false);
+        let strict_filter = |peer: &Peer| {
+            basic_filter(peer)
+                && peer.quality.connection_transient_fail_count == 0
+                && peer.quality.connection_attempt_count > 0
```

Comment:
this only means we tried to connect to the peer, but didn't necessarily succeed in doing so; wouldn't a check for `connected_count` - or some new metric like `connection_init_success_count` - be a better fit here?

Reply:
oh, I see - when a peer gets connected to, the value of `connection_transient_fail_count` is reset; that works too